python/ray/serve/long_poll.py

```diff
@@ -104,6 +113,17 @@ def _poll_next(self):
         self._current_ref._on_completed(
```

Comment:
we haven't changed the fact that we're overwriting `_on_completed` when ray client is used, can this be avoided ?

Reply:
That's a bigger change unfortunately